### PR TITLE
Backend Code review feedback implementation and write Unit test code for unit testing of File upload page

### PR DESCRIPTION
**Pull request by Anupma012**
**Implementation Details :**
Addressed the following review comments by Jamin and updates the code.
1.Use regex instead of split
2.Use file.type to fetch the file extension instead of splitting it
3.Use function to add country and without country logic.
4.When file upload to S3 fails display the error message in the UI.
5.Reducing the branches
6.If possible handle multiple exceptions with the same try block.
7.Writing unit test cases.
8.Use class based views instead function based views.
9.Use jQuery instead of JS.
10.Restrict uploading only excel files from the UI.

**Testing Details:**

1. Tested by uploading non .xlsx files, to check the error message.
2. Tested by uploading .xlsx files, to check the success message.
3. Tested by checking the file name format matches (ClientName_(Country/SplitName)_ClientCategory_Survey_Input_YYYYMMDD' (country name being optional)).
4. Tested if the file is being uploaded to s3 bucket successfully.
5. Checked if the category names are mapping to that of client names.
6. Checked if for a single client name if we have two categories, is it being displayed or not.
7.  Ensured that all unit tests pass successfully.

### CODE REVIEW

**Comment by sahilgupta71754** [comment]
Please resolve the merge conflict, also ask Abhishek to review and approve it

File: UI/interface/user/tests/test_input_file.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    def test_parse_s3_uri(self):
+        view = UploadHubInputFileView()
+        view.S3_URI = 's3://test-bucket/test-key'
+
+        bucket_name, key = view.parse_s3_uri()
+
+        self.assertEqual(bucket_name, 'test-bucket')
+        self.assertEqual(key, 'test-key')
+
+    def test_check_file_name_valid(self, file_name_parts, customer_name, category_name, file_type, file_name_format_split):
```
[the remark this answers is not in the document]
Comment (abhishek-p8) [reply UI/interface/user/tests/test_input_file.py:93]: If you are using the class variables then you can directly use them inside the method, no need to send it as arguments.

File: UI/interface/user/views.py [hunk 2]
```diff
@@ -88,134 +90,143 @@ def form_valid(self,form):
...
+            if data:
+                cursor = connection.cursor()
+                sql_query = "SELECT DISTINCT (client_category) FROM hap.dbo.client_info WHERE client_name=%s"
+                cursor.execute(sql_query, (data,))
+                rows = cursor.fetchall()
+                return Response({"category": rows}, status=status.HTTP_200_OK)
+            else:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+        except Exception as e:
+            logging.error(f"An error occurred: {e}")
```
Comment (abhishek-p8) [inline UI/interface/user/views.py:108]: don't hard code the error message

File: UI/interface/user/views.py [hunk 3]
```diff
@@ -88,134 +90,143 @@ def form_valid(self,form):
...
-        rows = cursor.fetchall()
-        return Response({"category": rows}, status=status.HTTP_200_OK)
-    else:
-        return Response(status=status.HTTP_404_NOT_FOUND)
+    def parse_s3_uri(self):
+        try:
+            parsed_url = urlparse(S3_URI[5:])
+            bucket_name, key = parsed_url.path.split('/', 1)
+            return bucket_name, key
+        except Exception as e:
```
Comment (abhishek-p8) [inline UI/interface/user/views.py:188]: Generic exception should be used when you have handled all other exceptions for the code block.

File: UI/interface/user/views.py [hunk 4]
```diff
@@ -88,134 +90,143 @@ def form_valid(self,form):
...
+            with hub_input_file_obj as file_obj:
+                s3_client.upload_fileobj(file_obj, bucket_name, f"{key}/{file_name_without_extension}")
+            return {'message': SUCCESS_MESSAGES['File_Upload_Success']}
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchBucket':
+                logging.error(ERROR_MESSAGES['Bucket_Failure'])
+            elif e.response['Error']['Code'] == 'AccessDenied':
+                logging.error(ERROR_MESSAGES['Bucket_Access_Denied'])
+            else:
+                logging.error(f'An unexpected error occurred: {e}')
```
Comment (abhishek-p8) [inline UI/interface/user/views.py:181]: Don't hard code error/success messages.

**Review by abhishek-p8 — APPROVED** [review]
Approved.

**Review by sahilgupta71754 — APPROVED** [review]
Approved

**Pull request merged**